stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -600,6 +600,9 @@ true
 """
 function norm(itr, p::Real=2)
     isempty(itr) && return float(norm(zero(eltype(itr))))
+    v, s = iterate(itr)
+    !isnothing(s) && !ismissing(v) && v == itr && throw(ArgumentError(
+        "cannot evaluate norm recursively if the initial element is identical to that of the container"))
```

Comment:
```suggestion
        "cannot evaluate norm recursively if the type of the initial element is identical to that of the container"))
```

Reply:
Note that the condition is an equivalence of objects, not types.

Reply:
Oh, I missed that. I thought the condition is the same as in the other PR, and as suggested by @jishnub. So why the difference?

Reply:
An 'infinite loop' happens when the first (and possible only) element keeps getting accessed e.g. `'c'`. We can't rely on types since e.g. `typeof(first('c')) == typeof('c')` because `first('c') == 'c'`. Sorry if that's unclear.

Reply:
> `typeof(first('c')) == typeof('c')`

That's why the type condition was introduced in the other PR and suggested earlier, wasn't it? I mean `typeof(v) == typeof(itr)`, to avoid calling recursively `generic_norm` when iterating doesn't yield a different type.

Reply:
Also, the values may match even when the types don't, but this won't be a recursive call.

Reply:
So what's the conclusion here? Should we change the check to types instead of values? I think it's the equal type case when we call `generic_norm` with the exact same signature and when we run into infinite recursion, isn't it?

Reply:
IIRC, an existing test failed when using the types version of the check, but can't get to it right now to recall it.

Reply:
It would be better to go with the types, as that is the correct condition. Worth checking which test fails with that, and why.

Reply:
This is the one that I identified failing `TestGeneric` (`generic.jl`) with `typeof(v) == typeof(itr)`:
```
    @test [[1,2, [3,4]], 5.0, [6im, [7.0, 8.0]]] ≈ [[1,2, [3,4]], 5.0, [6im, [7.0, 8.0]]]
```
which asks for `norm([[1, 2, [3, 4]], 5.0, [6im, [7.0, 8.0]]])`, and runs as
```
julia> v, s = iterate([[1, 2, [3, 4]], 5.0, [6im, [7.0, 8.0]]])
(Any[1, 2, [3, 4]], 2)

julia> typeof(v)
Vector{Any} (alias for Array{Any, 1})

julia> typeof([[1, 2, [3, 4]], 5.0, [6im, [7.0, 8.0]]])
Vector{Any} (alias for Array{Any, 1})
```